test/SharedFx.UnitTests/SharedFxTests.cs

```diff
@@ -1,14 +1,76 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.IO.Compression;
+using System.Net;
+using System.Reflection;
 using Newtonsoft.Json.Linq;
 using Xunit;
 
 namespace Microsoft.AspNetCore
 {
     public class SharedFxTests
     {
+
+        [Theory]
+        [MemberData(nameof(GetSharedFxConfig))]
+        public void BaselineTest(SharedFxConfig config)
+        {
+            var previousVersion = TestData.GetPreviousAspNetCoreReleaseVersion();
+            var url = $"https://dotnetcli.blob.core.windows.net/dotnet/aspnetcore/Runtime/" + previousVersion + "/aspnetcore-runtime-internal-" + previousVersion + "-win-x64.zip";
+            var zipName = "assemblies.zip";
+            var nugetAssemblyVersions = new Dictionary<string, Version>();
+
+            var root = TestData.GetDotNetRoot();
+            var dir = Path.Combine(root, "shared", config.Name, config.Version);
+
+            using (var testClient = new WebClient())
+            {
+                testClient.DownloadFile(url, zipName);
+            }
+
+            var zipPath = Path.Combine(AppContext.BaseDirectory, zipName);
+
+            if (!Directory.Exists("unzipped"))
+            {
+                ZipFile.ExtractToDirectory(zipPath, "unzipped");
+            }
+
+            var nugetAssembliesPath = Path.Combine(AppContext.BaseDirectory, "unzipped", "shared", config.Name, previousVersion);
+
+            string[] files = Directory.GetFiles(nugetAssembliesPath, "*.dll");
+            foreach (string file in files)
+            {
+                try
+                {
+                    var assemblyVersion = AssemblyName.GetAssemblyName(file)?.Version;
+                    var splitPath = file.Split('\\');
+                    var dllName = splitPath[splitPath.Length - 1];
+                    nugetAssemblyVersions.Add(dllName, assemblyVersion);
+                }
+                catch (BadImageFormatException) { }
+            }
+
+            files = Directory.GetFiles(dir, "*.dll");
+
+            Assert.All(files, file =>
+            {
+                try
+                {
+                    var localAssemblyVersion = AssemblyName.GetAssemblyName(file)?.Version;
+                    var splitPath = file.Split('\\');
+                    var dllName = splitPath[splitPath.Length - 1];
+                    Assert.True(nugetAssemblyVersions.ContainsKey(dllName), $"Expected {dllName} to be in the downloaded dlls");
```

Comment:
Oh and BTW just checking, are there are no xUnit asserts for these two asserts here? Usually `Assert.True` is frowned upon, unless there's no other reasonable choice.

Reply:
`Assert.Contains(nugetAssemblyVersions.Keys, ...)` should work here. The only supposed loss would be the custom error message when it fails.

Reply:
We could change to use the `Assert.Contains` api. The error message becomes slightly less clear. I'm indifferent.